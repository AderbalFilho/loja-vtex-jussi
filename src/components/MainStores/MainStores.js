import React from 'react';

import { JLink, JSection, JText } from '@ui';
import { LOGOS } from '@constants';

import './MainStores.scss';

function MainStores() {
  return (
    <JSection classes="main-stores">
      <div className="main-stores__container">
        <JText color="black" classes="main-stores__text">
          Nossas principais lojas VTEX
        </JText>
        <JText color="black" classes="main-stores__text main-stores__arrow">
          {window.innerWidth < 1024 ? '↓' : '→'}
        </JText>
        {LOGOS.map((logo) => {
          return (
            <JLink
              key={logo.imgAlt}
              href={logo.link}
              target="_blank"
              rel="noreferrer noopener"
            >
              <img
                src={logo.img1x}
                srcSet={`${logo.img1x}, ${logo.img2x} 2x, ${logo.img3x} 3x`}
                alt={logo.imgAlt}
                className="main-stores__logo-img"
              />
            </JLink>
          );
        })}
      </div>
    </JSection>
  );
}

export default MainStores;
