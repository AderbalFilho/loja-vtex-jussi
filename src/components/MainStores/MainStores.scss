.main-stores {
  padding: 21px 0;
}

.main-stores__container {
  align-items: center;
  display: flex;
  justify-content: space-between;
  margin: 0 auto;
  max-width: 1024px;
  width: 100%;
}

.main-stores__text {
  margin: 0;
}

.main-stores__arrow {
  font-size: 1.5rem;
  line-height: 0.5rem;
  margin-top: -6px;
}
