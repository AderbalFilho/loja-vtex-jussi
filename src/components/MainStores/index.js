import MainStores from './MainStores';

export default MainStores;
