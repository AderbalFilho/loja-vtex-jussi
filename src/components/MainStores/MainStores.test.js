import React from 'react';
import TestRenderer from 'react-test-renderer';

import MainStores from './MainStores';

describe('MainStores', () => {
  it('matches snapshot', () => {
    const testRenderer = TestRenderer.create(<MainStores />);
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });
});
