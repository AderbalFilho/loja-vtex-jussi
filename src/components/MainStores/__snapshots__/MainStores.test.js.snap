// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainStores matches snapshot 1`] = `
<section
  className="j-section j-section--white main-stores"
>
  <div
    className="main-stores__container"
  >
    <p
      className="j-text j-text--black j-text--normal main-stores__text"
    >
      Nossas principais lojas VTEX
    </p>
    <p
      className="j-text j-text--black j-text--normal main-stores__text main-stores__arrow"
    >
      â†’
    </p>
    <a
      className="j-link"
      href="https://www.brastemp.com.br/"
      rel="noreferrer noopener"
      target="_blank"
    >
      <img
        alt="Logo Brastemp"
        className="main-stores__logo-img"
        src=""
        srcSet=",  2x,  3x"
      />
    </a>
    <a
      className="j-link"
      href="https://www.compracerta.com.br/"
      rel="noreferrer noopener"
      target="_blank"
    >
      <img
        alt="Logo Compra Certa"
        className="main-stores__logo-img"
        src=""
        srcSet=",  2x,  3x"
      />
    </a>
    <a
      className="j-link"
      href="https://loja.consul.com.br/"
      rel="noreferrer noopener"
      target="_blank"
    >
      <img
        alt="Logo Consul"
        className="main-stores__logo-img"
        src=""
        srcSet=",  2x,  3x"
      />
    </a>
    <a
      className="j-link"
      href="https://br.thebar.com/"
      rel="noreferrer noopener"
      target="_blank"
    >
      <img
        alt="Logo The Bar"
        className="main-stores__logo-img"
        src=""
        srcSet=",  2x,  3x"
      />
    </a>
  </div>
</section>
`;
